google-sheets/trialSheet.py

```diff
@@ -0,0 +1,55 @@
+import os
+# imports
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+class GoogleSheetsAPI:
+    def __init__(self, credentials="token1.json", key ="./credentials.json" , SPREAD_ID = None):
+        self.SCOPES = ["https://www.googleapis.com/auth/spreadsheets"]  #accessed by multiple functions in multiple places 
+        self.credentials = credentials
+        self.key = key
+        self.SPREAD_ID = SPREAD_ID # we leave this parameter as none since whenever a new object is specified this will be unique identifier of it
+
+    def auth(self):
+        credentials = None
+        if os.path.exists(self.credentials):
+            credentials = Credentials.from_authorized_user_file(self.credentials, self.SCOPES)
+        if not credentials or not credentials.valid:
+            if credentials and credentials.expired and credentials.refresh_token:
+                credentials.refresh(Request())
+            else:
+                flow = InstalledAppFlow.from_client_secrets_file(self.key, self.SCOPES)
+                credentials = flow.run_local_server(port=0)
+            with open(self.credentials, "w") as token:
+                token.write(credentials.to_json())
+        self.service = build("sheets", "v4", credentials=credentials) #now this variable will be available in other method
+
+    def modify(self, SHEET_NAME):
```

Comment:
here you should pass more variables which we can you later in the code thus avoiding the hard coding